File: Fitness_App/backend/test-ai-version.js
import "dotenv/config";
import { GoogleGenerativeAI } from '@google/generative-ai';

async function test() {
  const genAI = new GoogleGenerativeAI(process.env.GEMINI_API_KEY);
  try {
     const model = genAI.getGenerativeModel({
          model: "gemini-flash-latest",
          // generationConfig: { responseMimeType: "application/json" }
     });
     const result = await model.generateContent("hello");
     console.log("Success gemini-flash-latest:", result.response.text());
  } catch(e) {
     console.error("Crash gemini-flash-latest:", e.message);
  }
}
test();

File: Fitness_App/backend/test_ai_version.py
import os
import asyncio
import google.generativeai as genai
from dotenv import load_dotenv

load_dotenv()

async def test():
    api_key = os.getenv("GEMINI_API_KEY")
    genai.configure(api_key=api_key)
    try:
        model = genai.GenerativeModel(model_name="gemini-flash-latest")
        result = await model.generate_content_async("hello")
        print("Success gemini-flash-latest:", result.text)
    except Exception as e:
        print("Crash gemini-flash-latest:", str(e))

if __name__ == "__main__":
    asyncio.run(test())
